uri/FactoryTest.php

```diff
@@ -62,6 +62,87 @@ public static function validFilePath(): array
         ];
     }
 
+    /** @dataProvider provideValidData */
+    public function testFromData(string $data, ?string $mimetype, ?string $parameters, string $expected): void
+    {
+        $arguments = [$data];
+        if (null !== $mimetype) {
+            $arguments[] = $mimetype;
+        }
+
+        if (null !== $parameters) {
+            $arguments[] = $parameters;
```

Comment:
this won't work well if `$mimetype` is passed as `null` and `$parameters` is provided, as it will actually send the parameters as the mimetype.

A solution would be to use `$arguments['parameters'] = $parameters;` to inject it as a named argument

Reply:
Currently, your `empty parameters` test in the data provider actually tests `empty mimetype`

Reply:
`null` is only for the tests... the signature prohibits `null` and only accepts `string`

Reply:
Updated the test to remove the confusion. No more null value.

Reply:
I know null was just for the test. But the implementation made it lie about what it was actually testing